### PR TITLE
[DEVOPS-942] Remove GitHub envs from lifecycle and simplify workflows

### PR DESCRIPTION
**Pull request by Aloutsis**
# Orfium GitHub Actions

## Description

Example Deployment: https://github.com/Orfium/todo/actions/runs/4552826161
Example Teardown: https://github.com/Orfium/todo/actions/runs/4552935505

## Checklist

- [x] I have reviewed my code and checked that there are no unrelated changes in this pull request
- [x] I have created a JIRA ticket describing the purpose of this pull request
- [x] I have checked if the changes to the templates are breaking or have contacted the DevOps team for feedback
- [x] I have updated the template with the Actionlint Format guidelines: [Actionlint](https://github.com/rhysd/actionlint#readme). Information on how to do that are provided on the CONTRIBUTING.md file under docs/ folder.

### CODE REVIEW

File: .github/workflows/teardown.yml [hunk 1]
```diff
@@ -52,20 +30,31 @@ jobs:
         with:
           python-version: "3.10"
 
+      - name: Check infra folder path
```
Comment (nVlast) [inline .github/workflows/teardown.yml:33]: Would it help you if you set a default value for `inputs.infra_folder_path`? It seems that you could erase this step completely.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -148,36 +103,36 @@ jobs:
...
+          cat ${{ steps.infra-folder-path.outputs.infra_folder_path }}samconfig.toml
           echo ${{ steps.combine-cf-parameters.outputs.SAM_PARAMETERS }}
 
           MAIN_TEMPLATE='${{ inputs.main_cf_template }}'
           if [[ -z $MAIN_TEMPLATE ]]; then
-            sam build -t ${{ needs.set-environment-variables.outputs.infra_folder_path }}aws-deploy.yml
+            sam build -t ${{ steps.infra-folder-path.outputs.infra_folder_path }}aws-deploy.yml
           else
             sam build -t ${{ inputs.main_cf_template }}
           fi
```
Comment (nVlast) [inline .github/workflows/deploy.yml:130]: Maybe we could avoid this part if we set a default value for `inputs.main_cf_template` :thinking:

File: .github/workflows/teardown.yml (hunk 1 above)
Comment (Aloutsis) [reply .github/workflows/teardown.yml:33]: Good question. I always forget about the default values. Let me check if it works and I'll update the PR if necessary.

**Review by nVlast — APPROVED** [review]
Looks good to me. Approved

**Review by mgiannos — APPROVED** [review]

**Pull request merged**